src/persistence/settings.cpp

```diff
@@ -1028,6 +1030,23 @@ void Settings::setDateFormat(const QString &format)
     dateFormat = format;
 }
 
+int Settings::getMarkdownPreference() const
+{
+    QMutexLocker locker{&bigLock};
+    return markdownPreference;
+}
+
+void Settings::setMarkdownPreference(int newValue)
+{
+    QMutexLocker locker{&bigLock};
+    if (newValue < 0)
+        newValue = 1;
+    else if (newValue > 2)
+        newValue = 2;
```

Comment:
If either of those branches are taken, this is an internal bug that we shouldn't silently ignore.
This wouldn't be as much of a problem with an enum instead of an int (use the type system!), but if you're going to check for internal interface violations, please assert() or at least log the bug.